torch/_dynamo/utils.py

```diff
@@ -758,7 +758,10 @@ def wrap_to_fake_tensor(e, fake_mode):
 
 
 def wrap_to_fake_tensor_and_record(e, tx):
-    if type(e) in (torch.Tensor, torch.nn.Parameter):
+    # The not fake tensor check here is annoying - ideally, fake tensors never call this during wrapping.
+    # However, get_fake_value takes args and passes them through this, which may include fake tensors.
+    # see tree_map(fake_wrapper, args) in get_fake_value.
```

Comment:
⚠️

Reply:
Agreed, but I do not want to shave the yak and rewrite huge parts of our logic. This is easier for now and we can flip this to an assert and burn it down later.

Reply:
I am probably forced to shave this yak for shape env rewrite